### ISSUE TITLE
An error in my first error lesson. 

### ISSUE THREAD

**Issue by clownstudios** [GDQuest/learn-gdscript#153]
**Describe the bug**
If I Rom in lesson 2 my first error IT give an error cant connect with sth. 

**To Reproduce**
Steps to reproduce the bug:
1. Do the lesson til my first error. Or just juse this link https://gdquest.github.io/learn-gdscript/#course/lesson-2-your-first-error/practice-85733.tres

**Expected behavior**
Nö plan

**Screenshots**
![Screenshot_20220119-152643_Samsung Internet](https://user-images.githubusercontent.com/68462312/150152558-f98d95e3-3d8c-4a19-ae16-efd574442af9.png)

**Information about your device (please complete the following information):**
 - Operating System: Android 11
 - Browser Samsung Internet

**Additional context**
Please Fix!! 

Thanks, 
Moritz

**Comment by NathanLovato** [comment]
It's a connection issue. This will happen when the server is offline (typically rebuilding) or you have an internet connection issue. @pycbouh is there something we can do about that UX-wise?

**Comment by pycbouh** [comment]
I think some indicator can be added to indicate if the connection is live or not. I think the concept of connection and connection problems is not too foreign for an average computing device user, so it should be clear what that means, even if we add some hint about language server not responding. I think @Xananax was working on some heartbeat system for the language server, so when it's ready, we can use it.

Also, I'd rephrase the message a bit, since we cannot provide any useful details. We shouldn't so openly use specific domain terminology and should give something more generic and, more importantly, more directed so that student has a suggestion about what to do next. We have "Are you connected?" but I think it's too ambiguous as to what and how one should be connected (connection to LSP is implicit, students have no idea).

**Comment by Xananax** [comment]
There are multiple messages now in `master`, depending on what kind of failure the connection is encountered: time out, error, CORS, etc. The entire connection logic and error handling are more robust. Some of those new errors have additional explanations and details through the error database. Not sure it helps users a lot though, indeed.

The heartbeat functionality that I was talking about was to test if the server is up from the docker, and send me/us a notification if it isn't. But it could be repurposed for a client-side "connected" UI LED. You can send a request to `http://lsp.gdquest.com/ping` every `n` seconds. The response should be a plain-text `pong`.

Though it's probably better to open a long-lived connection or socket, rather than polling. I could add that functionality in both the server and Godot, it shouldn't take too long.

**Comment by Xananax** [comment]
Commit 9a3dc6dc3bcf79413ac801edbbb2b84ad4d54cda adds a socket server, and commit https://github.com/GDQuest/gdscript-syntax-server/commit/5ecfd2479c8f47f1e4f18188f61088ad6b653e03 adds a socket server

Left to do:
- Server-side
  - [ ] specify the port for server through config
  - [ ] open the port from security rules for usage 
  - [ ] actually use the new module (it's for the moment separate)
-  Client-side
  - [ ] Specify the port (and the host?) through `project.godot`
  - [ ] Decide on a mechanism of usage (Autoload?)
  - [ ] Tie signals to some UI button that shows status (possibly with placebo clicking to reconnect?)

**Comment by NathanLovato** [comment]
I'm opening a separate issue for the new feature. The error message got tweaked and should be more explicit.
